lib/tasks/mail_tasks.rake

```diff
@@ -0,0 +1,17 @@
+namespace :mail_tasks do
+  desc "rake task for heroku schedule to send out the weekly notifications."
+  task send_weekly_notifications: :environment do
+    if DateTime.now.wday == 1   #Heroku schedule only allows for every day task not every week so this ensures its only sent of monday
+      week = Week.upcoming.first
+      assignments = Assignment.where(week_id: week.id)
+
+      assignments.each do |assignment|
+        employee = assignment.employee
+
+        UserMailer.assignment_reminder(employee).deliver
+       end
```

Comment:
Please put everything inside the `if sunday?` block into a new service called `SendsAssignmentReminders`. So the task is just:

```
SendsAssignmentReminders.call if sunday?
```

This is ideal because this rake task might be replaced by an ActiveJob, Sidekiq Worker, or any number of other task-runner objects. Also, we can decide to send reminders from anywhere in the app (i.e. maybe a manager wants to click a button to send an extra reminder).